amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -235,13 +247,17 @@ public void run() {
 
         try {
             initialize();
-            log.info("Initialization complete. Starting worker loop.");
+            log.info("Initialization complete. Scheduling periodicShardSync..");
```

Comment:
Reasoning for this not being in initialize():
1. PeriodicShardSync is scheduled (with an executor service) to run with a fixed delay. But for initialization to succeed, we need to sync shards at least one time so that we can spin up the leaseCoordinator. So we'll need a way to ensure that the first iteration of PeriodicShardSync succeeded, which makes things a bit messy. 
2. Clients can always _skipShardSyncAtWorkerInitializationIfLeasesExist_ to prevent every worker from syncing shards during initialization
3. initiazlie() runs _maxInitializationAttempts_ times. If scheduling of PeriodicShardSync  succeeds, and leaseCoordinator fails, we'd have to deal with stopping and restarting of PeriodicShardSync and the leaderElection tasks. (this is not a big deal, but something that adds additional logic)

Given these reasons, I decided to leave the existing shardSync logic as is, and schedule the leaderElected periodicShardSync after initialize(). But if you think it's worth the tradeoff, we can move it to initialize().

Reply:
Ok, then make sense.